src/components/Tabset/index.js

```diff
@@ -69,38 +135,54 @@ export default class Tabset extends Component {
         }
     }
 
-    isFirstTab() {
+    isLeftButtonDisabled() {
         const { activeTabName } = this.props;
-        const { tabChildren } = this.state;
+        const { tabChildren, screenWidth, tabsetScrollLeft } = this.state;
         const activeTabIndex = getActiveTabIndex(tabChildren, activeTabName);
-        if (activeTabIndex === 0) {
+        if (screenWidth < 600 && activeTabIndex === 0) {
+            return true;
+        } if (screenWidth > 600 && tabsetScrollLeft === 0) {
             return true;
         }
         return false;
     }
 
-    isLastTab() {
+    isRightButtonDisabled() {
         const { activeTabName } = this.props;
-        const { tabChildren } = this.state;
+        const { tabChildren, screenWidth, tabsetScrollLeft, tabsetMaxScroll } = this.state;
         const lastTabIndex = tabChildren.length - 1;
         const activeTabIndex = getActiveTabIndex(tabChildren, activeTabName);
-        if (lastTabIndex === activeTabIndex) {
+        if (screenWidth < 600 && lastTabIndex === activeTabIndex) {
+            return true;
+        } if (screenWidth > 600 && tabsetScrollLeft === tabsetMaxScroll) {
             return true;
         }
         return false;
     }
 
-    handleSelectNextTab() {
+    handleRightButtonClick() {
+        const { screenWidth, tabsetWidth, tabsetScrollLeft } = this.state;
+        if (screenWidth > 600) {
+            this.tabsetRef.current.scrollLeft = tabsetScrollLeft + tabsetWidth;
+            this.updateTabsetButtonsVisibility();
+            return null;
```

Comment:
you need to perform this, there is not need to run all updateTabsetButtonsVisibility function again since the only you need to do here is disable and enable buttons, no more change when a button is clicked
also you can:
```return this.updateTabsetButtonsVisibility();```
and remove:
```return null;```